optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/OptaPlannerProcessor.java

```diff
@@ -211,31 +214,77 @@ private void applySolverProperties(RecorderContext recorderContext,
-                        + SolverBuildTimeConfig.DEFAULT_SCORE_DRL_URL + " resource.");
+        }
+    }
+
+    private Optional<String> constraintsDrl() {
+        if (optaPlannerBuildTimeConfig.constraintsDrl.isPresent()) {
+            String constraintsDrl = optaPlannerBuildTimeConfig.constraintsDrl.get();
+            ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+            if (classLoader.getResource(constraintsDrl) == null) {
+                throw new IllegalStateException("Invalid " + OptaPlannerBuildTimeConfig.CONSTRAINTS_DRL_PROPERTY
+                        + " property (" + constraintsDrl + "): that classpath resource does not exist.");
             }
-            solverConfig.setScoreDirectorFactoryConfig(scoreDirectorFactoryConfig);
         }
+        return optaPlannerBuildTimeConfig.constraintsDrl;
+    }
+
+    private Optional<String> defaultConstraintsDrl() {
+        ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+        return classLoader.getResource(OptaPlannerBuildTimeConfig.DEFAULT_CONSTRAINTS_DRL_URL) != null
+                ? Optional.of(OptaPlannerBuildTimeConfig.DEFAULT_CONSTRAINTS_DRL_URL)
+                : Optional.empty();
+    }
+
+    private ScoreDirectorFactoryConfig defaultScoreDirectoryFactoryConfig(IndexView indexView, Optional<String> constrainsDrl) {
+        ScoreDirectorFactoryConfig scoreDirectorFactoryConfig = new ScoreDirectorFactoryConfig();
+        scoreDirectorFactoryConfig.setEasyScoreCalculatorClass(
+                findImplementingClass(DotNames.EASY_SCORE_CALCULATOR, indexView));
+        scoreDirectorFactoryConfig.setConstraintProviderClass(
+                findImplementingClass(DotNames.CONSTRAINT_PROVIDER, indexView));
+        scoreDirectorFactoryConfig.setIncrementalScoreCalculatorClass(
+                findImplementingClass(DotNames.INCREMENTAL_SCORE_CALCULATOR, indexView));
+        constrainsDrl.ifPresent(value -> scoreDirectorFactoryConfig.setScoreDrlList(Collections.singletonList(value)));
+        if (scoreDirectorFactoryConfig.getEasyScoreCalculatorClass() == null
+                && scoreDirectorFactoryConfig.getConstraintProviderClass() == null
+                && scoreDirectorFactoryConfig.getIncrementalScoreCalculatorClass() == null
+                && scoreDirectorFactoryConfig.getScoreDrlList() == null) {
+            throw new IllegalStateException("No classes found that implement "
+                    + EasyScoreCalculator.class.getSimpleName() + ", "
+                    + ConstraintProvider.class.getSimpleName() + " or "
+                    + IncrementalScoreCalculator.class.getSimpleName() + ".\n"
+                    + "Neither was a property " + OptaPlannerBuildTimeConfig.CONSTRAINTS_DRL_PROPERTY + " defined, nor a "
+                    + OptaPlannerBuildTimeConfig.DEFAULT_CONSTRAINTS_DRL_URL + " resource found.\n");
```

Comment:
I'd argue that this fail fast duplicates a fail-fast in SolverFactory.create(solverConfig).

Also, if we ever support SolverConfigCustomizer in Spring Boot, we can do these checks here yet, because the customer might "fix it".

Reply:
So far my philosophy has been that the Spring Boot starter and Quarkus extension should not duplicate fail fasts that
1) are detected as fail-fasts in SolverConfig later itself, once the SolverFActory is created
2) have a good error message.

Now, the 2) is debatable here, indeed.

Reply:
=> bluejeans topic?

Reply:
Agreed to leave as it is.